### PR TITLE
feat: Improves messenger's performance

### PR DESCRIPTION
**Pull request by soluchok**
Improves messenger's performance.
Messenger will not use storage.

Signed-off-by: Andrii Soluk <isoluchok@gmail.com>

### CODE REVIEW

**Review by fqutishat — APPROVED** [review]

File: pkg/didcomm/common/service/messenger.go [hunk 1]
```diff
@@ -31,8 +31,15 @@ type Messenger interface {
 	// ReplyTo replies to the message by given msgID.
 	// Keeps threadID in the *decorator.Thread.
 	// Using this function means that communication will be on the same thread.
+	//
+	// Deprecated: Please do not use it anymore. The function can be removed in future release.
```
Comment (DRK3) [inline pkg/didcomm/common/service/messenger.go:35]: Issue #?

File: pkg/client/messaging/client.go [hunk 2]
```diff
@@ -212,7 +212,7 @@ func (c *Client) Reply(ctx context.Context, msg json.RawMessage, msgID string, s
 	}
 
 	action = func() error {
-		return c.ctx.Messenger().ReplyTo(msgID, didCommMsg)
+		return c.ctx.Messenger().ReplyTo(msgID, didCommMsg) // nolint: staticcheck
```
Comment (DRK3) [inline pkg/client/messaging/client.go:215]: What was the linter complaining about? (asking out of curiosity)

File: pkg/didcomm/common/service/messenger.go [hunk 3]
```diff
@@ -64,5 +71,7 @@ type NestedReplyOpts struct {
 	TheirDID string
 	// MsgID to which nested reply to be sent,
 	// optional when all the above parameters are provided.
+	//
+	// Deprecated: Please do not use it anymore. The field can be removed in future release.
```
Comment (DRK3) [inline pkg/didcomm/common/service/messenger.go:75]: Issue #?

File: pkg/client/messaging/client.go (hunk 2 above)
Comment (soluchok) [reply pkg/client/messaging/client.go:215]: Deprecated usage.

**Review by DRK3 — APPROVED** [review]

**Review by fqutishat — APPROVED** [review]

**Pull request merged**